skyportal/models/public_pages/public_source_page.py

```diff
@@ -0,0 +1,160 @@
+__all__ = ['PublicSourcePage']
+
+import json
+import jinja2
+import sqlalchemy as sa
+from sqlalchemy import event
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.orm import deferred
+from baselayer.app.env import load_env
+from baselayer.app.json_util import to_json
+from baselayer.app.models import (
+    Base,
+    UserAccessControl,
+    CustomUserAccessControl,
+)
+from ..source import Source
+from ..group import GroupUser
+
+from ...utils.cache import Cache, dict_to_bytes
+
+env, cfg = load_env()
+
+cache_dir = "cache/public_pages/sources"
+cache = Cache(
+    cache_dir=cache_dir,
+    max_age=cfg["misc.minutes_to_keep_public_source_pages_cache"] * 60,
+)
+
+
+def published_source_access_logic(cls, user_or_token):
+    """Return a query that filters PublicSourcePage instances based on user access."""
+    # if the user is a system admin, he can update and delete all published sources
+    # otherwise, he can only delete and update the published sources associate with his group
+    user_id = UserAccessControl.user_id_from_user_or_token(user_or_token)
+    query = sa.select(cls)
+    if not user_or_token.is_system_admin:
+        query = query.join(Source, cls.source_id == Source.obj_id)
+        query = query.join(GroupUser, Source.group_id == GroupUser.group_id)
+        query = query.filter(GroupUser.user_id == user_id)
+    return query
+
+
+class PublicSourcePage(Base):
+    """Public page of a source on a given date."""
+
+    update = delete = CustomUserAccessControl(published_source_access_logic)
+
+    id = sa.Column(sa.Integer, primary_key=True)
+
+    source_id = sa.Column(sa.String, nullable=False, doc="ID of the source")
+
+    created_at_iso = sa.Column(
```

Comment:
Do we need to have a duplicate of `created_at`??? I would strongly advise against it. We already have that in the DB, pretty much in ISO format too.